lib/avromatic/model/custom_type.rb

```diff
@@ -0,0 +1,55 @@
+require 'avromatic/model/null_custom_type'
+
+module Avromatic
+  module Model
+
+    # Instances of this class contains the configuration for custom handling of
+    # a named type (record, enum, fixed).
+    class CustomType
+
+      attr_accessor :to_avro, :from_avro, :value_class
+
+      def initialize(value_class)
+        @value_class = value_class
+      end
+
+      # A coercer method is used when assigning to the model. It is used both when
+      # deserializing a model instance from Avro and when directly instantiating
+      # an instance. The coecer method must accept a single argument and return
+      # the value to store in the model for the attribute.
+      def coercer
```

Comment:
Should we call this `decoder` (or better yet `avro_decoder` and `avro_encoder` instead of `coder`)? It wasn't clear to me based on the name what `coercer` would do vs. `coder`.